Fix JS for strict mode syntax error

fredngo opened the pull request:
When javascript is running in strict mode, functions are required to be declared only at the top level or immediately within another function.

This pull request fixes the JS to be compatible with strict mode.

Here is the error as recorded from Firebug:

```
SyntaxError: in strict mode code, functions may be declared only at top level or immediately within another function
function load_new_items(){ 
... 
```

fredngo committed:
Fix JS for strict mode syntax error:

“SyntaxError: in strict mode code, functions may be declared only at top level or immediately within another function
function load_new_items(){
“

HeeL commented:
Thanks for the fix. If there is nothing else that has to be fixed for a strict mode, you can also add a use strict in the top

fredngo committed:
Fix var declaration for strict mode.

fredngo committed:
Add use strict directive

fredngo committed:
Properly remove the scroll event handler when the page doesn’t have a data-scrollinity-path element, using a namespaced jquery event. (See // https://css-tricks.com/namespaced-events-jquery/)

It is necessary to manually garbage-collect event handlers that are bound to top level objects such as $(window) when navigating pages with Ajax (such as with Turbolinks). If the event handler is not removed, then when the user scrolls to the bottom of the next page, it will continue to call the scrolling event handler in the background!

fredngo commented:
Hey, actually there were a couple more small bugs to fix with strict mode with the var declarations. I fixed them. And I added `use strict` at the top. 

I also fixed a fairly serious problem where the scroll handler continues to be attached to `$(window)` even after moving on to the next page, when the navigation is performed with Ajax. The scroll handler is now properly removed when the next page doesn't have `data-scrollinity-path` elements.

Could I ask you to release the updated gem to rubygems.org? Thanks!

HeeL commented:
Yes, thanks, I merged it and the new version will be released later today

fredngo commented:
Thanks!!
